lib/webhdfs/client_v1.rb

```diff
@@ -280,6 +280,7 @@ def request(host, port, method, path, op=nil, params={}, payload=nil, header=nil
       conn.open_timeout = @open_timeout if @open_timeout
       conn.read_timeout = @read_timeout if @read_timeout
 
+      path = URI.escape(path) # make path safe for transmission via HTTP
```

Comment:
Ruby's `URI.escape` is now obsolete.
Use another method. There's some option, so please select best one of them.